internal/storage/cache/cache.go

```diff
@@ -0,0 +1,66 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"go.flipt.io/flipt/internal/cache"
+	"go.flipt.io/flipt/internal/storage"
+)
+
+var _ storage.Store = &Store{}
+
+type Store struct {
+	storage.Store
+	cacher cache.Cacher
+}
+
+func New(store storage.Store, cacher cache.Cacher) storage.Store {
+	return &Store{store, cacher}
+}
+
+func (s *Store) SetCache(ctx context.Context, key string, value any) error {
+	cachePayload, err := json.Marshal(value)
+	if err != nil {
+		return err
+	}
+
+	return s.cacher.Set(ctx, key, cachePayload)
+}
+
+func (s *Store) GetCache(ctx context.Context, key string, value any) (bool, error) {
+	cachePayload, cacheHit, err := s.cacher.Get(ctx, key)
+	if err != nil || !cacheHit {
+		return cacheHit, err
+	}
+
+	err = json.Unmarshal(cachePayload, value)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (s *Store) GetEvaluationRules(ctx context.Context, namespaceKey, flagKey string) ([]*storage.EvaluationRule, error) {
+	cacheKey := fmt.Sprintf("s:get:er:%s:%s", namespaceKey, flagKey)
```

Comment:
```suggestion
	cacheKey := fmt.Sprintf("s:er:%s:%s", namespaceKey, flagKey)
```

don't think we need the `get` here, as the key represents:

`storage:evaluationRules:{namespace}:{flag}`